src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -50,6 +55,17 @@ public static void Initialize()
         // as you can't make a constructor partial.
         private RuntimeEventSource(int _) { }
 
+        private enum EventId : int
+        {
+            AppContextSwitch = 1
+        }
+
+        [Event((int)EventId.AppContextSwitch, Level = EventLevel.Informational, Keywords = Keywords.AppContext)]
+        internal void LogAppContextSwitch(string switchName, int value)
+        {
+            base.WriteEvent((int)EventId.AppContextSwitch, switchName, value);
+        }
+
         protected override void OnEventCommand(EventCommandEventArgs command)
         {
             if (command.Command == EventCommand.Enable)
```

Comment:
Does this enable event command still come through with startup tracing?
(for example `dotnet trace collect --providers System.Runtime:1 -- app.exe`)

Reply:
Good question, but out of my wheelhouse and I don't really know how to test it. @brianrob?

Reply:
`OnEventCommand` should be fired whenever an `EventSource` is enabled, and exists in the process.  So on start-up, when the `EventSource` comes into existence if `dotnet trace` has already started collection, then the `EventSource` will be registered with EventPipe, be told that it's enabled, and then call this code.  So the short answer (after the long answer) is yes.

Reply:
> how to test it

I'd think you could create an app that has some feature switch set in its runtimeconfig.json, set the `DOTNET_ROOT` environment variable to point to your local build, and then collect a trace with `dotnet trace collect --providers System.Runtime:1 -- app.exe`

Reply:
Looks like it works -- I got the AppContext switch out when I set it via `RuntimeHostConfigurationOption`